components/selection/selection-header.js

```diff
@@ -0,0 +1,186 @@
+				min-height: 36px;
+			}
+			d2l-selection-select-all {
+				flex: none;
+			}
+			d2l-selection-summary {
+				flex: none;
+				margin-left: 0.9rem;
+			}
+			:host([dir="rtl"]) d2l-selection-summary {
+				margin-left: 0;
+				margin-right: 0.9rem;
+			}
+			d2l-selection-select-all-pages {
+				flex: none;
+				margin-left: 0.45rem;
+			}
+			:host([dir="rtl"]) d2l-selection-select-all-pages {
+				margin-left: 0;
+				margin-right: 0.45rem;
+			}
+			.d2l-selection-header-actions {
+				--d2l-overflow-group-justify-content: flex-end;
+				flex: auto;
+				text-align: right;
+			}
+			:host([dir="rtl"]) .d2l-selection-header-actions {
+				text-align: left;
+			}
+			.d2l-sticky-edge {
+				margin-top: -1px;
+				position: absolute;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.noSelection = false;
+		this.noSticky = false;
+		this.selectAllPagesAllowed = false;
+		this._scrolled = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._stickyObserverDisconnect();
+	}
+
+	render() {
+		const classes = this._getSelectionHeaderContainerClasses();
+		return html`
+			<div class="d2l-sticky-edge"></div>
+			<div class="${classMap(classes)}">
+				${this.noSelection ? null : html`
+					<d2l-selection-select-all></d2l-selection-select-all>
+					<d2l-selection-summary
+						aria-hidden="true"
+						no-selection-text="${this.localize('components.selection.select-all')}"
+					>
+					</d2l-selection-summary>
+					${this.selectAllPagesAllowed ? html`<d2l-selection-select-all-pages></d2l-selection-select-all-pages>` : null}
+				`}
+				<div class="d2l-selection-header-actions">
+					<d2l-overflow-group opener-type="icon"><slot @slotchange="${this._handleSlotChange}"></slot></d2l-overflow-group>
+				</div>
+			</div>
+			${!this.noSticky ? html`<div class="d2l-selection-header-shadow"></div>` : null}
+		`;
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('noSticky')) {
+			this._stickyObserverUpdate();
+		}
+	}
+
+	_getSelectionHeaderContainerClasses() {
+		return {
+			'd2l-selection-header-container': true,
+			'd2l-selection-header-container-slim': (!this._hasActions && !this.selectAllPagesAllowed)
+		};
+	}
+
+	_handleSlotChange(e) {
+		this._hasActions = (e.target.assignedNodes({ flatten: true }).filter(node => node.nodeType === Node.ELEMENT_NODE).length > 0);
+	}
+
+	_stickyObserverDisconnect() {
+		if (this._stickyIntersectionObserver) {
+			this._stickyIntersectionObserver.disconnect();
+			this._stickyIntersectionObserver = undefined;
+			this._scrolled = false;
+		}
+	}
+
+	_stickyObserverUpdate() {
+		this._stickyObserverDisconnect();
+
+		if (!this.noSticky && typeof(IntersectionObserver) === 'function') {
+			this._stickyIntersectionObserver = new IntersectionObserver(([e]) => {
+				this._scrolled = !e.isIntersecting;
+			}, { threshold: 1 });
+
+			const target = this.shadowRoot.querySelector('.d2l-sticky-edge');
+			this._stickyIntersectionObserver.observe(target);
```

Comment:
This is mostly the same as the original (from `list.js`). I've added `threshold: 1` which tells the IntersectionObserver to watch for when 100% of the element is visible - though since it's an empty div, it doesn't really make a difference either way.

Reply:
I found that the `_scrolled` state was being triggered when scrolling left/right (i.e. when the left edge of the header was obscured).

I've fixed that by removing the `threshold: 1` here and adding `left`/`right` styles to the `.d2l-sticky-edge` - now the entire div needs to be hidden to trigger the `_scrolled` state.